src/App/Osu/OsuBeatmap.cpp

```diff
@@ -1626,11 +1626,17 @@ void OsuBeatmap::fail()
 
 	if (!m_osu->isInMultiplayer() && osu_drain_kill.getBool())
 	{
-		engine->getSound()->play(getSkin()->getFailsound());
+		if(osu_auto_reset.getBool())
+		{
+			restart();
+		}
+		else{
+			engine->getSound()->play(getSkin()->getFailsound());
 
-		m_bFailed = true;
-		m_fFailAnim = 1.0f;
-		anim->moveLinear(&m_fFailAnim, 0.0f, osu_fail_time.getFloat(), true); // trigger music slowdown and delayed menu, see update()
+			m_bFailed = true;
+			m_fFailAnim = 1.0f;
+			anim->moveLinear(&m_fFailAnim, 0.0f, osu_fail_time.getFloat(), true); // trigger music slowdown and delayed menu, see update()
+		}
```

Comment:
You are a beginner, but it should be easy to understand that when committing code to an existing repository you **must** match the existing coding style. ImAgInE If i WrOtE eNgLiSh LiKe ThIs ThE eNtIrE tImE, that's very annoying to read yes? This is what your code looks like to me. So please fix that.

For my personal coding style, you don't need `{}` brackets for one-line if-statements under certain conditions (there must either be an empty line below or an `else`, to avoid dangling-ifs and to make reading the code easier), so you can just have the top part like this:

`if (osu_auto_reset.getBool())`
`	restart();`
`else`
`{`
`	...`
`}`